lib/buffer.js

```diff
@@ -348,6 +348,21 @@ Buffer.concat = function(list, length) {
   return buffer;
 };
 
+Buffer.encode = function(source, encoding, targetEncoding) {
+  if (typeof source !== 'string')
+    throw new TypeError('"source" argument must be a string');
+  if (!targetEncoding) {
+    targetEncoding = encoding;
+    encoding = 'utf8';
+  }
+  targetEncoding = targetEncoding || 'utf8';
+  encoding = encoding || 'utf8';
```

Comment:
There's no handling if `encoding` or `targetEncoding` are `1` or `true`, etc. Tests don't cover it either. Also this should be placed in the `else` of the above `if` since both would have already been assigned.